module/VuFind/src/VuFind/Controller/MyResearchController.php

```diff
@@ -186,6 +186,35 @@ protected function processAuthenticationException(AuthException $e)
         $this->flashMessenger()->addMessage($msg, 'error');
     }
 
+    /**
+     * Store a referer (if appropriate) to keep post-login redirect pointing
+     * to an appropriate location. This is used when the user clicks the
+     * log in link from an arbitrary page or when a password is mistyped;
+     * separate logic is used for storing followup information when VuFind
+     * forces the user to log in from another context.
+     *
+     * @param bool  $allowCurrentUrl Whether the current URL is valid for followup
+     * @param array $extras          Extra data for the followup
+     *
+     * @return void
+     */
+    protected function setFollowupUrlToReferer(bool $allowCurrentUrl = true, array $extras = [])
+    {
+        // lbreferer is the stored current url of the lightbox
+        // which overrides the url from the server request when present
+        $refer = $this->getRequest()->getQuery()->get(
+            'lbreferer',
+            $this->getRequest()->getServer()->get('HTTP_REFERER', null)
+        );
+        $normReferer = $this->normalizeUrlForComparison($refer);
+        $myUserReset = $this->getServerUrl('myresearch-verify');
+        $murNorm = $this->normalizeUrlForComparison($myUserReset);
+        if (str_starts_with($normReferer, $murNorm)) {
+            return;
+        }
+        parent::setFollowupUrlToReferer($allowCurrentUrl, $extras);
```

Comment:
What is the reason for extending the method in this controller instead of adjusting the base method?

Reply:
added this code to the base method instead of extending.